stdlib/Profile/src/heapsnapshot_reassemble.jl

```diff
@@ -0,0 +1,163 @@
+# TODO(PR): This code hasn't been reviewed yet.
```

Comment:
Okay yeah, i just did some more sanity checking, and there are definitely still some bug(s) in the current reconstruction. It's still producing nonsensical edges. I suspect a bug in the string table reconstruction, but it needs some investigation.

Reply:
Okay, i've fixed the bugs in my latest push. :) Things look much more reasonable now. I'll address the rest of your comments next; thanks for leaving early feedback Jameson! ❤️